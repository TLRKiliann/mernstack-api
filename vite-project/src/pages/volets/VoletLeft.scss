/*Left volet*/
.volet--L {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 400px;
  min-height: 120vh;
  overflow: hidden;
}

@keyframes volet--leftdisplay {
  0% {
    transform: translateX(-400px);
    overflow: hidden;
  }
  100% {
    transform: translateX(0px);
    overflow: hidden;
  }
}

.subvolet--L {
  position: relative;
  top: 0px;
  left: 0px;
  width: -400px;
  min-height: 120vh;
  background-color: rgba(45, 85, 255, 0.4);
  border-right: 1px solid blue;
  transform: translateX(0px);

  animation-name: volet--leftdisplay;
  animation-duration: 1s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 11;
}

.subvolet--L button {
  margin-top: 10px;
  margin-left: 350px;
  width: 30px;
  height: 30px;

  border-radius: 50% 50%;
  background: transparent;
  color: whitesmoke;
  &:hover {
    cursor: pointer;
    background: rgba(0, 155, 255, 0.5);
    color: lightgreen;
  }
}

.subvolet--L p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background-color: rgba(45, 85, 255, 0.6);
  border: 2mm ridge royalblue;
  border-radius: 5px;
  font-weight: bold;
  color: whitesmoke;
}

.section--voletl {
  margin: 20px 20px;
  padding: 10px 40px;
  background-color: rgba(45, 85, 255, 0.6);
  border: 2mm ridge royalblue;
  border-radius: 10px;
}

.span {
  display: flex;
  align-items: center;
  justify-content: center;
}

.li {
  margin-bottom: 10px;
  padding-bottom: 10px;
  display: flex;
  align-items: center;
  font-weight: bold;
  border-bottom: 1px dashed whitesmoke;
}

.link--tagvoletl {
  margin: 2px 0px;
  text-decoration: none;
  font-family: 'Consolas';
  font-size: 1.2rem;
  color: whitesmoke;
  &:hover {
    color: steelblue;
  }
}

.volet--L2 {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 400px;
  min-height: 120vh;
  overflow: hidden;
}

.subvolet--L2 {
  position: relative;
  top: 0px;
  left: 0px;
  width: -400px;
  min-height: 120vh;
  background-color: rgba(45, 85, 255, 0.4);
  border-right: 1px solid blue;
  transform: translateX(-400px);

  animation-name: volet--lefthidden;
  animation-duration: 1s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 11;
}

@keyframes volet--lefthidden {
  0% {
    transform: translateX(0px);
    overflow: hidden;
  }
  100% {
    transform: translateX(-400px);
    overflow: hidden;
  }
}

.subvolet--L2 button {
  margin-top: 10px;
  margin-left: 350px;
  width: 30px;
  height: 30px;
  background: transparent;
  border-radius: 50% 50%;
  color: whitesmoke;
}

.subvolet--L2 p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background-color: rgba(45, 85, 255, 0.6);
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 3px whitesmoke;
  font-weight: bold;
  color: whitesmoke;
}