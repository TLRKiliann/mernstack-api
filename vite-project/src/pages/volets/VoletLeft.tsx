import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { MdConnectWithoutContact } from 'react-icons/md'
import { TbPlugConnected } from 'react-icons/tb'
import { BsFillChatDotsFill } from 'react-icons/bs'
import './VoletLeft.scss'


interface VoletLeftProps {
  isOpenL: boolean
  handleVoletsLeft: () => void
}

const VoletLeft: React.FC = (props: VoletLeftProps) => {

  const [searchUser, setSearchUser] = useState<Array<string>>([])
  const [userFound, setUserFound] = useState<Array<string>>([])
  //console.log(searchUser)

  const handleSearchInput = (e: React.ChangeEvent<HTMLIputElement>): void => {
    setSearchUser(e.target.value)
  }

  const handleSearchUser = (e: React.MouseEvent<HTMLButtonElement>) => {
    const findByFirstName = props.users?.map(user => user).filter(user => {
      return user.firstName === searchUser
        ? `${user.firstname} ${user.lastName}`
        : null
    })
    e.preventDefault();
    if (searchUser === "") {
      setSearchUser("");
      alert("User undefined or not existing...")
    } else {
      setUserFound(findByFirstName)
      setSearchUser("")
    }
  };

  return(
    <div>
      {props.isOpenL ? (
        <div className="volet--L">
          <div className="subvolet--L">
            <button 
              className="btn--volet"
              onClick={props.handleVoletsLeft}>X</button>
            <p>Private Chat</p>
              
            <span className="span">
              <BsFillChatDotsFill size={64}/>
            </span>

            <div className="form--search">
              <input
                type="text"
                value={searchUser}
                onChange={(e) => handleSearchInput(e)}
                placeholder="Search by firstname"
                className="search--input"
              />
              <button
                onClick={handleSearchUser}
                className="btn--search"
              >
                Enter
              </button>
            </div>

            
            <div className="result--search">
            {userFound?.map(userFirst => (
              <p key={userFirst.id}
                className="result--searchp">{userFirst.firstName} {userFirst.lastName} 
                <span 
                  style={userFirst.isConnected ? {color: 'lightgreen'} 
                    : {color: 'orange'}}
                >
                  &nbsp;{userFirst.isConnected ? "Connected" : "No Connected"}
                </span>
              </p>
            ))}
            </div>


            <section className="section--voletl">
              {props.users?.map(user => (
                <li key={user.id} className="li"
                  style={user.isConnected 
                  ? {color: 'lightgreen'}
                  : {color: 'orange'}}
                >
                  {user.isConnected ? (
                    <MdConnectWithoutContact
                      size={24}
                      style={{
                        marginRight: '10px',
                      }}
                    />
                    ) : (
                    <TbPlugConnected
                      size={24}
                      style={{marginRight: '10px'}}
                    />
                    )
                  }
                  <Link className="link--tagvoletl"
                    to={`/computerroom/privatemessage/${user.firstName}`}
                  >
                    {user.firstName} {user.lastName}
                  </Link>
                </li>
                ))
              }
            </section>
          
          </div>
        </div>
        
        ) : (

        <div className="volet--L2">
          <div className="subvolet--L2">
            <button
              className="btn--volet"
              onClick={props.handleVoletsLeft}>X</button>
            <p>Users State Connection</p>

            <span className="span">
              <BsFillChatDotsFill size={64}/>
            </span>

            <div className="form--search">
              <input
                type="text"
                value={searchUser}
                onChange={(e) => handleSearchInput(e)}
                placeholder="Search by firstname"
                className="search--input"
              />
              <button
                onClick={handleSearchUser}
                className="btn--search"
              >
                Enter
              </button>
            </div>

            <section className="section--voletl">
              {props.users?.map(user => (
                <li key={user.id} className="li">
                  {user.isConnected ? (
                    <MdConnectWithoutContact
                      size={24}
                      style={{marginRight: '10px'}}
                    />
                    ) : (
                    <TbPlugConnected
                      size={24}
                      style={{marginRight: '10px'}}
                    />
                    )
                  }
                  <Link className="link--tagvoletl"
                    to={`/computerroom/privatemessage/${user.firstName}`}
                  >
                    {user.firstName} {user.lastName}
                  </Link>
                </li>
                ))
              }
            </section>

          </div>
        </div>
        )
      }
    </div>
  )
}

export default VoletLeft