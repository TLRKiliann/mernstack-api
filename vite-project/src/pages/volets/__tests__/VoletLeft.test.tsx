<<<<<<< HEAD
import React from "react";
import { screen, render, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { expect, vi } from 'vitest'
import '@testing-library/jest-dom'
import VoletLeft from "../VoletLeft.tsx";
import { handleSearchUser } from '../VoletLeft.tsx'

test('MatchSnapShot test ComputerRoom', () => {
  const tree = create(<VoletLeft />)
  expect(tree.toJSON()).toMatchSnapshot();
});

test("function handleSearchUser testing", () => {
  const beverage = "Emrick"
  const handleSearchUser = vi.fn(todo => todo)
  handleSearchUser(beverage)
  render(
    <button
      data-testid="btn-testvoletleft"
      onClick={handleSearchUser}>
      Enter
    </button>
  )
  fireEvent.click(screen.getByTestId("btn-testvoletleft"))
  expect(handleSearchUser).toHaveReturnedWith("Emrick")
  expect(handleSearchUser).toHaveBeenCalledTimes(2)
});
=======
import React from "react"
import { screen, render } from '@testing-library/react'
import { test, expect, vi } from 'vitest'
import '@testing-library/jest-dom'
import VoletLeft from "../VoletLeft.tsx"
import handleSearchInput from '../VoletLeft'
import handleSearchUser from '../VoletLeft'

test('renders users state connection link', () => {
  render(<VoletLeft />)
  const linkElement = screen.getByText(/private chat/i)
  expect(linkElement).toBeInTheDocument()
})

test('Volet-L handleSearchInput test', () => {
  const funcInput = handleSearchInput
  expect(funcInput).toBeDefined()
})

test('Volet-L handleSearchUser test', () => {
  const funcSearch = handleSearchUser
  expect(funcSearch).toBeDefined()
})
>>>>>>> master
