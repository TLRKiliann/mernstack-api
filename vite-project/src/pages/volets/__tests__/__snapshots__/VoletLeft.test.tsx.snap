// Vitest Snapshot v1

exports[`MatchSnapShot test ComputerRoom 1`] = `
<div>
  <div
    className="volet--L2"
  >
    <div
      className="subvolet--L2"
    >
      <button
        className="btn--volet"
      >
        X
      </button>
      <p>
        Private Chat
      </p>
      <span
        className="span"
      >
        <svg
          fill="currentColor"
          height={64}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width={64}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8c0 3.866-3.582 7-8 7a9.06 9.06 0 0 1-2.347-.306c-.584.296-1.925.864-4.181 1.234-.2.032-.352-.176-.273-.362.354-.836.674-1.95.77-2.966C.744 11.37 0 9.76 0 8c0-3.866 3.582-7 8-7s8 3.134 8 7zM5 8a1 1 0 1 0-2 0 1 1 0 0 0 2 0zm4 0a1 1 0 1 0-2 0 1 1 0 0 0 2 0zm3 1a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"
          />
        </svg>
      </span>
      <div
        className="form--search"
      >
        <input
          className="search--input"
          onChange={[Function]}
          placeholder="Search by firstname"
          type="text"
          value={[]}
        />
        <button
          className="btn--search"
          onClick={[Function]}
        >
          Enter
        </button>
      </div>
      <section
        className="section--voletl"
      />
    </div>
  </div>
</div>
`;
